src/quick-lint-js/error.h

```diff
@@ -1028,6 +1034,13 @@
           QLJS_TRANSLATABLE("for-of loop expression cannot have semicolons"),  \
           semicolon))                                                          \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_unexpected_semicolon_after_else, "E202",                           \
```

Comment:
Blocking: Someone else snagged E202. Pick a different error code.

Reply:
Moved the warning about unexpected behavior due to semicolon to E204 and else has empty body to E205.